core/ArchiveProcessor/Loader.php

```diff
@@ -429,4 +437,37 @@ public static function getArchivingDepth()
     {
         return self::$archivingDepth;
     }
+
+    private function shouldForceInvalidatedArchive($value, $tsArchived)
+    {
+        $params = $this->params;
+
+        // the archive is invalidated and we are in a browser request that is allowed archive it
+        if ($value == ArchiveWriter::DONE_INVALIDATED
+            && !Rules::isArchivingDisabledFor([$params->getSite()->getId()], $params->getSegment(), $params->getPeriod()->getLabel())
+        ) {
+            // if coming from core:archive, force rearchiving, since if we don't the entry will be removed from archive_invalidations
+            // w/o being rearchived
+            if (SettingsServer::isArchivePhpTriggered()) {
+                return true;
+            }
+
+            // if coming from a browser request, and period does not contain today, force rearchiving
+            if (!$params->getPeriod()->isDateInPeriod(Date::factory('today'))) {
```

Comment:
do we need to add the timezone to `today` depending on the idsite maybe?

Reply:
:+1: